### PR TITLE
Retry job update after failure to prevent modification conflict

### LINKED ISSUE #34585: [k8s.io] Job should fail a job {Kubernetes e2e suite}
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/33763/kubernetes-pull-build-test-gci-e2e-gce/1852/

Failed: [k8s.io] Job should fail a job {Kubernetes e2e suite}

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/job.go:209
Expected error:
    <*errors.StatusError | 0xc8208a5b80>: {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {SelfLink: "", ResourceVersion: ""},
            Status: "Failure",
            Message: "Operation cannot be fulfilled on jobs.batch \"foo\": the object has been modified; please apply your changes to the latest version and try again",
            Reason: "Conflict",
            Details: {Name: "foo", Group: "batch", Kind: "jobs", Causes: nil, RetryAfterSeconds: 0},
            Code: 409,
        },
    }
    Operation cannot be fulfilled on jobs.batch "foo": the object has been modified; please apply your changes to the latest version and try again
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/job.go:205
```

Happened on a presubmit run in #33763.

Previous issues for this test: #28773 #29506 #30699 #32734

### PR DESCRIPTION
**Pull request by soltysh**
This fixes #34585 flake.

@janetkuo || @kubernetes/sig-apps  ptal

I've been getting too many emails recently wrt to that issue, so I wanted to "clean" my inbox a bit 😉

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37077)

### CODE REVIEW

**Comment by soltysh** [comment]
With this being a flake fix I'm tagging this as a 1.5 milestone, fyi @saad-ali @dims

**Comment by kargakis** [comment]
One comment, LGTM in general

**Comment by dims** [comment]
@k8s-bot verify test this

**Comment by soltysh** [comment]
Nit addressed, applying label based on previous approval.

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -3388,6 +3389,30 @@ func UpdateStatefulSetWithRetries(c clientset.Interface, namespace, name string,
 	return statefulSet, pollErr
 }
 
+type updateJobFunc func(*batch.Job)
+
+func UpdateJobWithRetries(c clientset.Interface, namespace, name string, applyUpdate updateJobFunc) (job *batch.Job, err error) {
+	jobs := c.Batch().Jobs(namespace)
+	var updateErr error
+	pollErr := wait.Poll(10*time.Millisecond, 1*time.Minute, func() (bool, error) {
```
Comment (soltysh) [inline test/e2e/framework/util.go]: Hmmm... shouldn't we use it for other *WithRetries here as well?

**Comment by soltysh** [comment]
Re-applying label back, I've updated bazel.

**Pull request merged**